pkg/network/domainspec/generators.go

```diff
@@ -275,3 +291,99 @@ func (b *MacvtapLibvirtSpecGenerator) discoverDomainIfaceSpec() (*api.Interface,
 		},
 	}, nil
 }
+
+type PasstLibvirtSpecGenerator struct {
+	vmiSpecIface *v1.Interface
+	domain       *api.Domain
+	vmi          *v1.VirtualMachineInstance
+}
+
+func (b *PasstLibvirtSpecGenerator) Generate() error {
+	// remove passt interface from domain spec devices interfaces
+	foundDomainInterface := false
+	for i, iface := range b.domain.Spec.Devices.Interfaces {
+		if iface.Alias.GetName() == b.vmiSpecIface.Name {
+			b.domain.Spec.Devices.Interfaces = append(b.domain.Spec.Devices.Interfaces[:i], b.domain.Spec.Devices.Interfaces[i+1:]...)
+			foundDomainInterface = true
+			break
+		}
+	}
+	if !foundDomainInterface {
+		return fmt.Errorf("failed to find interface %s in vmi spec", b.vmiSpecIface.Name)
+	}
+
+	ports := b.generatePorts()
+	args := append([]string{"--runas", "107", "-e"}, ports...)
+	cmd := exec.Command("/usr/bin/passt", args...)
+	cmd.SysProcAttr = &syscall.SysProcAttr{
+		AmbientCaps: []uintptr{unix.CAP_NET_BIND_SERVICE},
+	}
+
+	// connect passt's stderr to our own stdout in order to see the logs in the container logs
+	var reader io.ReadCloser
+	reader, err := cmd.StderrPipe()
+	if err != nil {
+		log.Log.Reason(err).Error("failed to get passt stderr")
+		return err
+	}
+	go func() {
+		const bufferSize = 1024
+		const maxBufferSize = 512 * bufferSize
+		scanner := bufio.NewScanner(reader)
+		scanner.Buffer(make([]byte, bufferSize), maxBufferSize)
+		for scanner.Scan() {
+			log.Log.Info(fmt.Sprintf("passt: %s", scanner.Text()))
+		}
+		if err = scanner.Err(); err != nil {
+			log.Log.Reason(err).Error("failed to read passt logs")
+		}
+	}()
+
+	err = cmd.Start()
+	if err != nil {
+		log.Log.Reason(err).Error("failed to start passt")
+		return err
+	}
+
+	err = cmd.Wait()
+	if err != nil {
+		log.Log.Reason(err).Error("failed waiting for passt going to background")
+		return err
+	}
```

Comment:
I find all of this to be passt specific code that has nothing to do with specifying the domain; this starts the passt process, and captures its logging iiuc.

Can you move this code to a dedicated passt pkg ? This file is not a junkyard for binding specific code - it will become one soon if we allow it to be.

Reply:
passt process is eventually generating the domain spec so it makes some sense to have it here.
But I get your point. Since we currently have only three entry points from `plugPhase2` (domain.Generate, dhcp.Generate, dhcp.Run), we will have to introduce another entry point.

Let's discuss it with the team and I will introduce the "cleanup" in anothrer PR.